src/machine.rs

```diff
@@ -413,6 +416,18 @@ impl<'mir, 'tcx> Evaluator<'mir, 'tcx> {
             preemption_rate: config.preemption_rate,
             report_progress: config.report_progress,
             since_progress_report: 0,
+            external_so_lib: config.external_so_file.as_ref().map(|lib_file_path| {
+                // Note: it is the user's responsibility to provide a correct SO file.
+                // WATCH OUT: If an invalid/incorrect SO file is specified, this can cause
+                // undefined behaviour in Miri itself!
+                (
```

Comment:
What is the point of these parentheses? They seem entirely redundant.

Reply:
`external_so_lib` is an `Option<(libloading::Library, std::path::PathBuf)>`; these parens are to make the tuple

Reply:
Ah, I didn't read the code properly, sorry for this.